Fix for double escaping content in nested editables

Comandeer opened the pull request:
<!--
🚨 If you want to submit a PR for a security vulnerability, please contact us directly
at https://ckeditor.com/contact/ instead. 🚨
-->
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [x] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

```
* [#4060](https://github.com/ckeditor/ckeditor4/issues/4060): Fixed: content inside nested editables was escaped twice.
```

## What changes did you make?

It seems that the content inside nested editables was changed due to the fact that it was escaped twice – once during widget initialization and second time during processing the whole editor content. The fix forces unescaping the content before the widget initialization, which eliminated the second, unnecessary escape.

## Which issues does your PR resolve?

Closes #4060.
<!-- Closes #<ANOTHER_ISSUE_NUMBER>. -->

jacekbogdanski reviewed (changes requested):
1. The testing button works incorrectly. It always gives green feedback for editing mode and red for source editing with and without widget changes.

![switch](https://user-images.githubusercontent.com/10279586/91410753-abbf8980-e847-11ea-92f8-fc90de619c2d.gif)

2. Switching between editing and source mode a second time gives a bit different HTML:
For the first switch:
```html
<div class="testwidget"><div class="content">CKEditor rulez<br /><script src="data:text/javascript,''"></script></div></div>
```
For the second switch:
```html
<div class="testwidget"><div class="content">CKEditor rulez<script src="data:text/javascript,''"></script></div></div>
```
Not sure if it's caused by widget definition itself, but it changes widget editable content:

![widget](https://user-images.githubusercontent.com/10279586/91413489-79b02680-e84b-11ea-914c-99b5ecc81974.gif)

3. I have been really worried about the performance impact of this change. However, it seems not that huge, a couple of ms for complex editable. Nevertheless, it would be nice to prepare some complex widget with a couple of editable elements to make sure everything works ok, WDYT?

Comandeer commented:
> The testing button works incorrectly. It always gives green feedback for editing mode and red for source editing with and without widget changes.

Actually it was intended to work only in WYSIWYG mode… But thanks to your feedback, I greatly simplified manual test – not it's basically just clicking a button :sweat\_smile:.

> Switching between editing and source mode a second time gives a bit different HTML

Yup, it was connected with invalid widget definition. Fixed.

> I have been really worried about the performance impact of this change. However, it seems not that huge, a couple of ms for complex editable. Nevertheless, it would be nice to prepare some complex widget with a couple of editable elements to make sure everything works ok, WDYT?

Added :+1:

jacekbogdanski reviewed (approved):
Good job!
